packages/framework/src/types/discover.types.ts

```diff
@@ -1,43 +1,40 @@
-import { ValidateFunction } from 'ajv';
-
 import { ActionStepEnum, ChannelStepEnum } from '../constants';
-import { Schema } from './schema.types';
-import { ActionStepOptions } from './step.types';
+import { JsonSchema, Schema } from './schema.types';
+import { StepOptions } from './step.types';
 import { Execute, WorkflowOptions } from './workflow.types';
+import { Awaitable } from './util.types';
 
 export type StepType = `${ChannelStepEnum | ActionStepEnum}`;
 
-export type Validate = ValidateFunction;
-
 export type DiscoverProviderOutput = {
   type: string;
   code: string;
-  resolve: (stepInputs: unknown) => unknown | Promise<unknown>;
+  resolve: (inputs: unknown) => Awaitable<unknown>;
   outputs: {
-    schema: Schema;
-    validate: Validate;
+    schema: JsonSchema;
+    unknownSchema: Schema;
```

Comment:
Here we add the new `unknownSchema`, an internal framework property that stores the unknown schema from a user-provided library. We use this when calling the `validate` function to ensure that all provided custom logic in validation is preserved.